woocommerce/woocommerce-ios
# [IAP] M2 Full Feature List screen

Closes: #10190 

## Description
This PR adds the `View Full Feature List` button in the M2 plan selection view, as well as opens a modal displaying a full feature list per plan.

Original Figma design:

![screen 2](https://github.com/woocommerce/woocommerce-ios/assets/3812076/d487c1db-ba2e-4a3b-a662-2e118433416d)

## Known differences:
There's two known differences that we haven't tackled, mainly because increments complexity. We'll loop with design and update these if are worth the change on a different PR. These are:

* The navigation bar has a different color (white vs grey): There are no built-in methods to change the color of a SwiftUI NavigationView. A possible alternative is to create some sort of `NavigationConfigurator` struct that conforms to `UIViewControllerRepresentable`, this can take a closure that accepts a `UINavigationController` and modifies its appearance.
* Some line items are ordered differently: All essential features appear first within a group, while "performance features only" appear latter. As an example:

| Design Proposal | Implementation |
|--------|--------|
| <img width="363" alt="Screenshot 2023-07-18 at 12 23 36" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/4d0d858e-02c1-4a14-ba59-13515211e1a1"> | <img width="365" alt="Screenshot 2023-07-18 at 12 23 42" src="https://github.com/woocommerce/woocommerce-ios/assets/3812076/e9a8b813-15aa-4f6e-82a2-3a0f7760f2ee"> |

## Testing instructions
On a Free Trial site, eligible for IAP (a new site can be created in Menu > Switch Sites > Add New Site) tap on Purchase Now and observe that:
* Scrolling below the Performance plan, there's a `View Full Feature List`.
* Tapping on the button shows the new "Full Feature List screen" in a modal view. Confirm some details:
  * Each group has a title
  * Each group has a list of features
  * Some of these features are marked with a star icon that represents "performance plan only"
  * Some of these features are marked with a 1 or 2 subscript, for example Print Shipping Labels
  * There are line separators between groups
  * The 1 and 2 subscripts appear at the bottom of the view, explaining more details about these features.

| - | - |
|--------|--------|
| ![Simulator Screen Shot - iPhone 11 Pro - 2023-07-18 at 12 13 21](https://github.com/woocommerce/woocommerce-ios/assets/3812076/cbf2e781-5a34-461b-aab6-599b4248e885) | ![Simulator Screen Shot - iPhone 11 Pro - 2023-07-18 at 12 13 27](https://github.com/woocommerce/woocommerce-ios/assets/3812076/014eebc4-554f-4686-9dd0-9f553ae3c900) | 

## Screenshots
![Simulator Screen Recording - iPhone 11 Pro - 2023-07-18 at 11 59 11](https://github.com/woocommerce/woocommerce-ios/assets/3812076/dbd1500f-ef55-4d0e-b195-f685e200ec83)
